src/components/App/App.js

```diff
@@ -15,51 +18,239 @@ import "./App.css";
 import "../../index.css";
 
 export default function App() {
-  // const [isLoggedIn, setIsLoggedIn] = useState(false);
-
+  const navigate = useNavigate();
+  // LOGIN & USER STATES
+  const [isLoggedIn, setIsLoggedIn] = useState(false);
+  const [currentUser, setCurrentUser] = useState({});
+  const [token, setToken] = useState(localStorage.getItem("jwt"));
+  // POPUP STATES
   const [isLoginPopupOpen, setIsLoginPopupOpen] = useState(false);
   const [isRegisterPopupOpen, setIsRegisterPopupOpen] = useState(false);
   const [isSuccessPopupOpen, setIsSuccessPopupOpen] = useState(false);
 
+  const [registerError, setRegisterError] = useState(false);
+
+  // NEWS STATES
+  const [isLoading, setIsLoading] = useState(false);
+  const [articles, setArticles] = useState([]);
+  const [savedArticles, setSavedArticles] = useState([]);
+
+  const [keyword, setKeyword] = useState('');
+
+  const [wasSearch, setWasSearch] = useState(false)
+
+  // CHECK TOKEN
+  useEffect(() => {
+    if (token) {
+      mainApi
+        .checkToken(token)
+        .then((res) => {
+          if (res.data._id) {
+            setCurrentUser(res.data);
+            setIsLoggedIn(true);
+          } else {
+            localStorage.removeItem("jwt");
+          }
+        })
+        .catch((err) => console.log(err));
+    }
+  }, [token]);
+
+  // FETCH SAVED ARTICLES ACCORDING TO USER
+  useEffect(() => {
+    if (isLoggedIn) {
+      mainApi.getArticles(token).then((res) => {
+        if (res.data)
+        setSavedArticles(res.data);
+      });
```

Comment:
You need to `catch` possible errors at the very end of any server request.

**Please, correct it everywhere**